cupy/cuda/nccl.pyx

```diff
@@ -145,6 +227,76 @@ cdef class NcclCommunicator:
     def __dealloc__(self):
         self.destroy()
 
+    @staticmethod
+    def initAll(int ndev, list devlist=None):
+        """ Initialize NCCL communicators for multiple devices in a single
+        process.
+
+        Args:
+            ndev (int): Number of GPUs to be used.
+            devlist (None or list): A list of integers that label the GPUs to
+                be used. The default is ``None``, meaning that the first
+                ``ndev`` GPUs will be chosen.
+
+        Returns:
+            list: A list of ``NcclCommunicator`` instances, each of which
+                controls one device.
+
+        .. note::
+            This method is for creating NCCL communicators in a single process
+            like this:
+
+            .. code-block:: python
+
+                from cupy.cuda import nccl
+                # Use 3 GPUs: #0, #2, and #3
+                comms = nccl.NcclCommunicator.initAll(3, [0, 2, 3])
+                assert len(comms) == 3
+
+            In a multi-process setup, use the default initializer instead.
+
+        .. seealso:: `ncclCommInitAll`_
+
+        .. _ncclCommInitAll:
+            https://docs.nvidia.com/deeplearning/sdk/nccl-developer-guide/docs/api/comms.html#ncclcomminitall
+        """  # noqa
+        cdef int i
+        cdef list comms = []
+        cdef NcclCommunicator NcclComm
```

Comment:
Would you use a variable name starting with a lower case here as well, how about `comm` for `NcclCommunicator` in contrast to `ncclCom` for `ncclComm_t`?

Reply:
OK will do.